custom_components/ha_washdata/suggestion_engine.py

```
@@ -605,20 +686,60 @@ def __init__(
+        # second is pure end-detection lag.  Tick just past the p95 update gap -
+        # matching DEFAULT_WATCHDOG_INTERVAL's documented "2 x sampling + 1"
+        # derivation for a publish-on-change sensor that skips at most one
+        # sample.  The old 3x multiple polled ~6x slower than the sensor updates
+        # and delayed end detection for no safety benefit.
+        # Floor at 2 x median + 1 so the suggestion pre-satisfies reconciler
+        # Rule 3a (watchdog >= 2 x sampling_interval) when CONF_SAMPLING_INTERVAL
+        # is suggested from the same median_dt.  Without this, a regular sensor
+        # (p95 ≈ median) would produce ceil(p95)+1 which Rule 3a then silently
+        # overwrites, leaving a stored value whose reason text no longer matches.
+        suggested_watchdog = int(max(30, max(math.ceil(p95_dt) + 1,
+                                            2 * math.ceil(median_dt) + 1)))
```

Ruff (0.16.1)

[warning] Nested `max` calls can be flattened

Flatten nested `max` calls

(PLW3301)